### PR TITLE
[mach] Fix Keychain handling on macOS Sierra

### PR DESCRIPTION
**Pull request by KrauseFx**
Fixes https://github.com/fastlane/fastlane/issues/5649

### CODE REVIEW

File: match/spec/utils_spec.rb [hunk 1]
```diff
@@ -23,12 +24,25 @@
...
+        expect(File).to receive(:exist?).with("#{Dir.home}/Library/Keychains/my/special.keychain-db").and_return(false)
         expect(File).to receive(:exist?).with('/my/special.keychain').and_return(false)
+        expect(File).to receive(:exist?).with('/my/special.keychain-db').and_return(false)
 
         expect do
           Match::Utils.import('item.path', '/my/special.keychain')
         end.to raise_error(/Could not locate the provided keychain/)
       end
+
+      it "tries to find the macOS Sierra keychain too" do
```
Comment (asfalcone) [inline match/spec/utils_spec.rb]: 👍

File: match/lib/match/utils.rb [hunk 2]
```diff
@@ -9,9 +9,18 @@ def self.import(item_path, keychain)
       # as the keychain path.
       #
       # We need to expand each path because File.exist? won't handle directories including ~ properly
+      #
+      # We also try to append `-db` at the end of the file path, as with Sierra the default Keychain name
+      # has changed for some users:
+      # - https://github.com/fastlane/fastlane/issues/5649
```
Comment (asfalcone) [inline match/lib/match/utils.rb]: Do we need to link to all of these issues in the comments? I feel like adding all of these here clutters up the code quite a bit.

**Review by asfalcone — APPROVED** [review]

File: match/lib/match/utils.rb (hunk 2 above)
Comment (KrauseFx) [inline match/lib/match/utils.rb]: I personally think it's good to provide some context and background on why a certain line or method was added. This way, future fastlane team can see why some things were added, including the complete discussion of why this change is necessary, with an easy way to actively reach out to people who had this problem. 
I agree with you that 3 issues might be too many, we can reduce it to the most important one, if you prefer that 👍

Comment (KrauseFx) [inline match/lib/match/utils.rb]: Whops, 2 are duplicates anyway, let me update the PR 👍

**Comment by KrauseFx** [comment]
Updated 👍

**Pull request merged**